libraries/AC_Avoidance/AP_OADijkstra.cpp

```diff
@@ -260,7 +260,14 @@ bool AP_OADijkstra::check_inclusion_polygon_updated() const
 bool AP_OADijkstra::create_inclusion_polygon_with_margin(float margin_cm, AP_OADijkstra_Error &err_id)
 {
     const AC_Fence *fence = AC_Fence::get_singleton();
+    
+    // skip unnecessary retry to build inclusion polygon if previous fence points have not changed 
+    if (_inclusion_polygon_update_ms == fence->polyfence().get_inclusion_polygon_update_ms()) {
+        return false;
+    }
+
     if (fence == nullptr) {
+        _inclusion_polygon_update_ms = fence->polyfence().get_inclusion_polygon_update_ms();
```

Comment:
Any reason not to set this once at the top of the function rather than before each return?

Reply:
At the time of writing this, I wrongly thought there was some logical problem in doing it this way. But on more thinking, it seems completely logical and obviously better to do this! I have updated the PR. Thanks!